evap/static/ts/src/confirmation-modal.ts

```diff
@@ -0,0 +1,62 @@
+import { selectOrError } from "./utils.js";
+
+export class ConfirmationModal extends HTMLElement {
+    static formAssociated = true;
+
+    readonly dialog: HTMLDialogElement;
+    readonly dialogForm: HTMLFormElement;
+    readonly type: string;
+    readonly internals: ElementInternals;
+
+    constructor() {
+        super();
+
+        const template = selectOrError<HTMLTemplateElement>("#confirmation-modal-template").content;
+        const shadowRoot = this.attachShadow({ mode: "open" });
+        shadowRoot.appendChild(template.cloneNode(true));
+
+        this.type = this.getAttribute("type") ?? "button";
+        this.internals = this.attachInternals();
+
+        this.dialog = selectOrError("dialog", shadowRoot);
+        this.dialogForm = selectOrError("form[method=dialog]", this.dialog);
+
+        const confirmButton = selectOrError("[data-event-type=confirm]", this.dialog);
+        const confirmButtonExtraClass = this.getAttribute("confirm-button-class") ?? "btn-primary";
+        confirmButton.className += " " + confirmButtonExtraClass;
+
+        selectOrError("[slot=show-button]", this).addEventListener("click", () => this.dialog.showModal());
+        this.dialogForm.addEventListener("submit", this.onDialogFormSubmit);
+    }
+
+    onDialogFormSubmit = (event: SubmitEvent) => {
+        event.preventDefault();
+
+        this.closeDialogSlowly();
+
+        if (event.submitter?.dataset?.eventType === "confirm") {
+            this.dispatchEvent(new CustomEvent("confirmed", { detail: new FormData(this.dialogForm) }));
+
+            if (this.type === "submit") {
```

Comment:
I haven't checked, but I think that this is how a `<button type=submit>` with a `click` event handler behaves, so I thought this would be the expected behavior. In addition, unlike buttons, the confirmation modal never has an implicit `type=submit`, which rules out unexpected submission. What do you think we should do?

Reply:
In-person-discussion: To prevent silly copy+paste errors, the modal shouldn't submit a form _and_ emit the event (which could in the end possibly submit two requests).

I don't really care how we solve it -- moving the `dispatchEvent` here into an else branch would be fine with me, your async-modal-idea also sounds fine.

Reply:
The `custom-success` / `reload-on-success` attributes I added work nicely, I kept the `confirmed` event in case we ever want to use the modal without a form